backend/services/table_anomaly_service.py

```
@@ -0,0 +1,433 @@
+                """, (table_name,))
+                pk_columns = [row[0] for row in cursor.fetchall()]
+                
+                if not pk_columns:
+                    return 0
+                
+                # Primary Key 기준으로 중복 검사
+                pk_columns_str = ', '.join(pk_columns)
+                cursor.execute(f"""
+                    SELECT COUNT(*) - COUNT(DISTINCT {pk_columns_str}) 
+                    FROM public.{table_name}
+                """)
```

Ruff (0.13.1)

Possible SQL injection vector through string-based query construction

(S608)